opcua/server/history_sql.py

```diff
@@ -231,6 +234,25 @@ def read_event_history(self, source_id, start, end, nb_values, evfilter):
     def _get_table_name(self, node_id):
         return str(node_id.NamespaceIndex) + '_' + str(node_id.Identifier)
 
+    def _get_event_fields(self, evtypes):
+        """
+        Get all fields from the event types that are to be historized
+        Args:
+            evtypes: List of event type nodes
+
+        Returns: List of fields for all event types
+
+        """
+        # get all fields from the event types that are to be historized
+        ev_aggregate_fields = []
+        for event_type in evtypes:
+            ev_aggregate_fields.extend((events.get_event_properties_from_type_node(event_type)))
+
+        ev_fields = []
+        for field in set(ev_aggregate_fields):
+            ev_fields.append(field.get_display_name().Text.decode(encoding='utf-8'))
+        return ev_fields
```

Comment:
Correct!

On Wed, Jun 15, 2016, 18:31 Andrew <notifications@github.com> wrote:

> In opcua/server/history_sql.py
> <https://github.com/FreeOpcUa/python-opcua/pull/207#discussion_r67197131>:
>
> > +        Get all fields from the event types that are to be historized
> > +        Args:
> > +            evtypes: List of event type nodes
> > +
> > +        Returns: List of fields for all event types
> > +
> > +        """
> > +        # get all fields from the event types that are to be historized
> > +        ev_aggregate_fields = []
> > +        for event_type in evtypes:
> > +            ev_aggregate_fields.extend((events.get_event_properties_from_type_node(event_type)))
> > +
> > +        ev_fields = []
> > +        for field in set(ev_aggregate_fields):
> > +            ev_fields.append(field.get_display_name().Text.decode(encoding='utf-8'))
> > +        return ev_fields
>
> I did this a while ago... I think the set set(ev_aggregate_fields) is
> what removes any duplicates.
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/FreeOpcUa/python-opcua/pull/207/files/83d14db5ccd3d520b931e1b340bfd64ab7f94948#r67197131>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ACcfzilQAsaw0Lry-pxToiZS5j4ZVt7Rks5qMCjzgaJpZM4I0hFH>
> .
>